homeassistant/components/obihai/config_flow.py

```diff
@@ -0,0 +1,116 @@
+"""Config flow to configure the Obihai integration."""
+from __future__ import annotations
+
+from types import MappingProxyType
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import CONF_OBIHAI_HOST, DEFAULT_PASSWORD, DEFAULT_USERNAME, DOMAIN
+from .obihai_api import validate_auth
+
+
+@callback
+def async_get_schema(
+    defaults: dict[str, Any] | MappingProxyType[str, Any]
+) -> vol.Schema:
+    """Return Obihai schema."""
+    schema = {
+        vol.Required(CONF_HOST, default=defaults.get(CONF_OBIHAI_HOST, "")): str,
+        vol.Optional(
+            CONF_USERNAME,
+            description={
+                "suggested_value": defaults.get(CONF_USERNAME, DEFAULT_USERNAME)
+            },
+        ): str,
+        vol.Optional(
+            CONF_PASSWORD,
+            default=defaults.get(CONF_PASSWORD, DEFAULT_PASSWORD),
+        ): str,
+    }
+
+    return vol.Schema(schema)
```

Comment:
It is easier to make the schema a constant (with system defaults), and then use `self.add_suggested_values_to_schema`

See
https://github.com/home-assistant/core/blob/cab8a59be48a027e2da9ed96aa782f7018d91b59/homeassistant/components/sfr_box/config_flow.py#L61